src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1598,12 +1603,58 @@ def process_gallery_image_version_namespace(cmd, namespace):
                     raise CLIError("usage error: {} is an invalid target region argument. "
                                    "The second part must be a valid integer replica count.".format(t))
 
+            # Parse target region encryption, example: ['des1,0,des2,1,des3', 'null', 'des4']
+            encryption = None
+            if hasattr(namespace, 'target_region_encryption') and namespace.target_region_encryption:
+                terms = namespace.target_region_encryption[i].split(',')
+                # OS disk
+                os_disk_image = terms[0]
+                if os_disk_image == 'null':
+                    os_disk_image = None
+                else:
+                    des_id = _disk_encryption_set_format(cmd, namespace, os_disk_image)
+                    os_disk_image = OSDiskImageEncryption(disk_encryption_set_id=des_id)
+                # Data disk
+                if len(terms) > 1:
+                    data_disk_images = terms[1:]
+                    data_disk_images_len = len(data_disk_images)
+                    if data_disk_images_len % 2 != 0:
+                        raise CLIError('usage error: LUN and disk encryption set for data disk should appear '
```

Comment:
in pair separating by comma like `lun1,disk_encrytption_setid1.lun2,disk_encryption_set2`

Reply:
Do you mean the error message can be polished?